### ISSUE TITLE
Including range (start and end) value in the generated JSON   

### ISSUE THREAD

**Issue by alfie-far** [tmccombs/hcl2json#36]
Hi, thanks for building such a great utility for HCL
We are currently using this library to perform static analysis checks on Terraform HCL files and we wanted to use the range (start and end position of tokens) info in order to point to the location of the resource in terraform template. I was wondering if you could help me find a solution for this issue

**Comment by amin-RFarzin** [comment]
That's true @tmccombs, it could be outside the scope of this project but I was thinking if you could help me realise where in the code I can access the position (_start and end range_) metadata in the code?

>Are you doing the static analysis with go, or outputting to json and doing static analysis with a different language?
We're using a wrapper library: [@cdktf/hcl2json](https://www.npmjs.com/package/@cdktf/hcl2json) in Node runtime to run our static analysis on the generated JSON file.

In terms of output format, I think adding position metadata requires creating an Abstract Syntax Tree so that we can keep metadata for each token of the HCL template.

**Comment by tmccombs** [comment]
Basically the way it works is [`ConvertFile`](https://github.com/tmccombs/hcl2json/blob/7220684b0cdc7dca4e5acdb228960fab091ec9bb/convert/convert.go#L57) Recursively walks through the AST of an [`hclsyntax.Body`](https://pkg.go.dev/github.com/hashicorp/hcl/v2@v2.10.1/hclsyntax#Body), and each part of the AST has one or more "Range" properties that describe the byte indices for where a section starts and ends. 

I hope that helps.

**Comment by amin-RFarzin** [comment]
Thank you @tmccombs!
That does help, I'll probably fork the project and start making changes to see if I can keep range info as a metadata of the generated json file.
